ocp_utilities/infra.py

```diff
@@ -49,6 +53,36 @@ def assert_nodes_schedulable(nodes):
         )
 
 
+def assert_pods_failed_or_pending(pods: list) -> None:
+    """
+    Validates all pods are not in failed nor pending phase
+
+    Args:
+         pods: List of pod objects
+
+    Raises:
+        PodsFailedOrPendingError, if there are failed or pending pods
+    """
+    LOGGER.info("Verify all pods are not failed nor pending.")
+
+    failed_or_pending_pods = [
+        (pod.name, pod.instance.status.phase)
```

Comment:
`pod.instance.status.phase` this call the API and you call it twice
Please convert to fol loop to reduce API calls